packages/deprecation-crawler/src/lib/processors/grouping.ts

```diff
@@ -0,0 +1,184 @@
+        group.matchers.push(parsedRegex);
+      }
+    } else {
+      groups.push({
+        key: groupKey || ungrouped,
+        matchers: parsedRegex !== '' ? [parsedRegex] : [],
+      });
+    }
+
+    deprecationsWithGroup.push({
+      ...deprecation,
+      group: groupKey,
+    });
+  }
+
+  updateRepoConfig({ ...config, groups });
+  return deprecationsWithGroup;
+}
+
+function getGroupNames(
+  groups: { key: string }[],
+  ungroupedKey: string
+): string[] {
+  return [
+    ungroupedKey,
+    ...groups.map((g) => g.key).filter((k) => k !== ungroupedKey),
+  ];
+}
+
+function checkForExistingGroup(groups: Group[], deprecation: Deprecation) {
+  return groups.find((group) => {
+    // If matchers are present test them else return false
+    return group.matchers.length
+      ? group.matchers.some((reg) =>
+          testMessage(reg, deprecation.deprecationMessage)
+        )
+      : false;
+  })?.key;
+}
+
+async function getGroupNameFromExistingOrInputQuestion(
+  deprecation: Deprecation,
+  groups: string[],
+  newGroupChoice = 'Create new group'
+): Promise<string> {
+  const answerNameFromExisting: { existingKey: string } = await prompt([
+    getGroupNameFromExistingQuestion(
+      deprecation,
+      [newGroupChoice, ...groups],
+      newGroupChoice
+    ),
+  ]);
+  const isExistingGroup = answerNameFromExisting.existingKey !== newGroupChoice;
+  return isExistingGroup
+    ? answerNameFromExisting.existingKey
+    : await prompt([getGroupNameQuestion(deprecation)]).then(
+        (s: { key: string }) => s.key
+      );
+}
+
+function getGroupNameFromExistingQuestion(
+  deprecation: Deprecation,
+  groupChoices: string[],
+  defaultKey: string
+) {
+  return {
+    // TODO: use autocomplete here? https://github.com/enquirer/enquirer/tree/master/examples/autocomplete
+    // @Notice Problem: can't have a custom input that is not in choices
+    type: 'select',
+    name: 'existingKey',
+    message:
+      `Add human readable group name to deprecation` +
+      EOL +
+      `${deprecation.path}#${deprecation.lineNumber}` +
+      EOL +
+      deprecation.deprecationMessage +
+      EOL,
+    initial: defaultKey,
+    choices: groupChoices,
+  };
+}
+
+function getGroupNameQuestion(deprecation: Deprecation) {
+  return {
+    type: 'input',
+    name: 'key',
+    message:
+      `Add human readable group name to deprecation` +
+      EOL +
+      `${deprecation.path}#${deprecation.lineNumber}` +
+      EOL +
+      deprecation.deprecationMessage +
+      EOL,
+    initial: ungrouped,
+  };
+}
+
+function getGroupRegexQuestion() {
+  return {
+    type: 'input',
+    name: 'regexp',
+    message:
+      `Which part of the deprecation message do you want to use as a matcher?` +
+      EOL +
+      ` Hint: regex string is allowed too.`,
+  };
+}
+
+function testMessage(reg: string, deprecationMessage: string): boolean {
+  return (
+    reg &&
+    new RegExp(parseDeprecationMessageOrRegex(reg)).test(
```

Comment:
@BioPhoton I also had to add `parseDeprecationMessageOrRegex` around the regex to make the tests pass

Reply:
What does it mean?

Reply:
Without it, the `whitespace-normalisation` and `all-lowercase` matchers didn't work.
The comment was normalized, but the regex did not.

Reply:
Yes. I left a comment in the code. Strange... Isn't it there.